docs/standard/threading/overview-of-synchronization-primitives.md

```diff
@@ -1,174 +1,167 @@
+
+### ReaderWriterLockSlim class
+
+Use the <xref:System.Threading.ReaderWriterLockSlim?displayProperty=nameWithType> class to grant exclusive access to a shared resource for writing. Multiple threads can access the resource simultaneously for reading. You might want to use <xref:System.Threading.ReaderWriterLockSlim> to synchronize access to a shared data structure that supports thread-safe read operations, but requires exclusive access to perform write operation. When a thread requests exclusive access (for example, by calling the <xref:System.Threading.ReaderWriterLockSlim.EnterWriteLock%2A?displayProperty=nameWithType> method), subsequent reader requests block until all existing readers have exited the lock, and the writer has entered and exited the lock.
   
- For a conceptual overview, see [Interlocked Operations](../../../docs/standard/threading/interlocked-operations.md).  
+For more information, see the [Reader-Writer locks](reader-writer-locks.md) article and the <xref:System.Threading.ReaderWriterLockSlim> API reference.
+
+### Semaphore and SemaphoreSlim classes
+
+Use <xref:System.Threading.Semaphore?displayProperty=nameWithType> or <xref:System.Threading.SemaphoreSlim?displayProperty=nameWithType> to limit the number of threads that can access a resource concurrently. Additional threads that request the resource wait until any thread releases the semaphore.
+
+You create a <xref:System.Threading.Semaphore> instance that represents a named system semaphore by using one of the [Semaphore constructors](<xref:System.Threading.Semaphore.%23ctor%2A>) that specifies a name or the <xref:System.Threading.Semaphore.OpenExisting%2A?displayProperty=nameWithType> method. A named system semaphore can be used for the inter-process synchronization.
+
+<xref:System.Threading.SemaphoreSlim> is a lightweight alternative to <xref:System.Threading.Semaphore> and can be used only for synchronization within a single process boundary. <xref:System.Threading.SemaphoreSlim> doesn't support named system semaphores.
+
+Neither of the semaphore classes has thread affinity. That means that one thread can acquire the semaphore and another one can release it.
   
+For more information, see the [Semaphore and SemaphoreSlim](semaphore-and-semaphoreslim.md) article and the <xref:System.Threading.Semaphore> or <xref:System.Threading.SemaphoreSlim> API reference.
+
+## Thread interaction, or signaling
+
+Thread interaction occurs when a thread must wait for notification, or signal, from one or more threads in order to proceed. For example, if a thread A calls the <xref:System.Threading.Thread.Join%2A?displayProperty=nameWithType> method of a thread B, the thread A is blocked until the thread B completes. The synchronization primitives described in the preceding section is another example of signaling: by releasing a lock a thread notifies another thread that it can proceed by acquiring the lock.
```

Comment:
- Thread interaction occurs when --> Thread interaction (or thread signaling) means that
- signal --> a signal
- a thread A --> thread A
- a thread B --> thread B
- the thread A --> thread A
- the thread B --> therad B
- is another example of of signaling --> provide a different mechanism for signaling
- comma after "by releasing a lock"